src/compile/facet.ts

```diff
@@ -245,7 +245,7 @@ export class FacetModel extends ModelWithField {
         as.push(`distinct_${field}`);
       }
     } else {
-      for (const channel of ['x', 'y'] as ScaleChannel[]) {
+      for (const channel of [X, Y] as ScaleChannel[]) {
```

Comment:
POSITION_SCALE_CHANNNELS?

Reply:
Done. I also replaced a few more places I found in the code.